src/zmqsubscriber/zmqsubscriber.go

```diff
@@ -59,92 +46,68 @@ func NewZMQSubscriber(host string, port string) (*ZMQSubscriber, error) {
-		Host:           host,
-		Port:           port,
-		Topics:         topics,
+		topics:         topics,
 		IncomingTx:     incomingTx,
 		IncomingBlocks: incomingBlocks,
 		socket:         socket,
 		cancel:         false,
 	}, nil
 }
 
-func (z *ZMQSubscriber) processMessage(topic string, payload [][]byte) error {
-	// TODO: use GetTime() and allow other time sources (eg NTP-corrected)
-	t := time.Now().UTC()
-
-	switch topic {
-	case TOPIC_HASHTX:
-		tx, err := parseTransaction(t, payload)
-		if err != nil {
-			return err
-		}
-		z.IncomingTx <- *tx
-	case TOPIC_RAWBLOCK:
-		block, err := parseBlock(t, payload)
-		if err != nil {
-			return err
-		}
-		z.IncomingBlocks <- *block
-	default:
-		return fmt.Errorf("unknown topic %s", topic)
-	}
-
-	return nil
-}
-
-// Listen for new messages, parse messages into native data types and put them into channels
-// `IncomingTx` and `IncomingBlocks`.
-// Returns error if something goes wrong and `nil` if stopped using `Stop()`
+// Run starts receiving new ZMQ messages. These messages are parsed according to
+// their topic and passed as native data types into the corresponding channels
+// (`IncomingTx` or `IncomingBlocks`). Run returns an error if an error occurs
+// while parsing. On normal stops with `Stop()` `nil` is returned.
 func (z *ZMQSubscriber) Run() error {
 	defer func() {
 		if err := z.socket.Close(); err != nil {
-			log.Printf("[ZMQ] socket closed with error %q (ignored)", err)
+			log.Printf("ZMQ subscriber socket closed with error (ignored): %s\n", err)
 		}
 	}()
 
 	parseErrors := make(chan error)
 
-	// FIXME(#11):  Workaround for bug where zmq crashes when Close() is called from a different
-	// 				goroutine. Instead of permanently blocking on Recv(), we set a timeout and check
-	// 				for `z.cancel`.
+	if err := z.socket.SetRcvtimeo(time.Second); err != nil {
+		return fmt.Errorf("could not set a receive timeout: %s", err)
+	}
+
+	// FIXME(#11):
+	// Workaround for a zmq crash when Close() is called from a different
+	// goroutine. Instead of permanently blocking on Recv(), a timeout is set and
+	// we check for `z.cancel`.
 	for !z.cancel {
-		// if a parse error happened, exit the loop
 		select {
 		case err := <-parseErrors:
 			return err
 		default:
 		}
 
-		if err := z.socket.SetRcvtimeo(time.Second); err != nil {
-			return fmt.Errorf("error setting timeout: %s", err)
-		}
-
 		msg, err := z.socket.RecvMessageBytes(0)
 		if err != nil {
-			if err == zmq4.ETIMEDOUT || err == zmq4.Errno(syscall.EAGAIN) || err == zmq4.Errno(syscall.EINTR) {
+			if err == zmq4.Errno(syscall.EAGAIN) {
+				log.Println("No ZMQ message received in the last second.")
+				continue
+			} else if err == zmq4.Errno(syscall.EINTR) {
 				continue
 			}
-
-			return fmt.Errorf("Could not receive ZMQ message: %s %v", err, err)
+			return fmt.Errorf("could not receive ZMQ message: %s", err)
 		}
 
 		topic, payload := string(msg[0]), msg[1:]
-		log.Printf(`[ZMQ] received topic "%s"`, topic)
```

Comment:
the idea here was to have prefixes for different subsystems that can be easily grepped

Reply:
Could use https://golang.org/pkg/log/#SetPrefix or set the `Lshortfile` flag (https://golang.org/pkg/log/#pkg-constants) for a short filename in the log.